lib/tapioca/gem/listeners/methods.rb

```diff
@@ -73,15 +73,29 @@ def compile_method(tree, symbol_name, constant, method, visibility = RBI::Public
           return unless method_owned_by_constant?(method, constant)
           return if @pipeline.symbol_in_payload?(symbol_name) && !@pipeline.method_in_gem?(method)
 
+          is_alias = method.name != method.original_name
           signature = lookup_signature_of(method)
-          method = T.let(signature.method, UnboundMethod) if signature
+          parameters = T.let(nil, T.nilable(T::Array[[Symbol, T.nilable(Symbol)]]))
+
+          if signature
+            method = T.let(signature.method, UnboundMethod)
+          elsif is_alias && signature.nil? && constant.method_defined?(method.original_name)
+            alias_source_method = constant.instance_method(method.original_name)
```

Comment:
In general, this is incorrect since alias binding happens at call site and doesn't get mutated. See: https://bsky.app/profile/fxn.bsky.social/post/3lbehp5rtes2j

```ruby
class A
  def foo = 42

  alias :bar :foo
end

class B < A
  def foo = 1
end

B.new.foo # => 1
B.new.bar # => 42
```